packages/astro/components/Debug.astro

```diff
@@ -0,0 +1,304 @@
+      <span class="value value-promise">Promise</span>
+    </>
+  }
+
+  if (type === 'array') {
+    value = node.length;
+    open = <><span class="none">Array</span>{'['}</>;
+    close = ']';
+  } else if (type === 'object') {
+    const keys = Object.keys(node);
+    if (keys.length === 0) {
+      value = 'Empty';
+    } else if (keys.length > 3) {
+      value = '…';
+    } else {
+      value = keys.slice(0, 3).join(',');
+    }
+    open = '{';
+    close = '}';
+  };
+
+  return <>
+    {key && <><span class="key">{key}</span>: </>}
+    {open && <span class="punc">{open}</span>}
+    <span class="hide">
+      <span class="len">{value}</span>
+      {close && <span class="punc">{close}</span>}
+    </span>
+  </>;
+};
+
+const Details = ({ node, key, children }) => {
+  const type = getType(node);
+  const props = {};
+
+  if (type === 'array' || type === 'object') {
+    props['data-char'] = type === 'array' ? ']' : '}'
+    props.open = !key && type === 'object' ? '' : undefined;
+  }
+  
+  return (
+    <details {...props}>
+        <Summary node={node} key={key} />
+        {children}
+    </details>
+  );
+}
+
+const Summary = ({ node, key }) => {
+  return (
+    <summary>{getSummary(node, key)}</summary>
+  );
+}
+
+const Empty = Symbol('Empty');
+
+const KeyValue = ({ key, value, dim }) => {
+  let type = key === '__proto__' ? 'prototype' : getType(value);
+  if (type === 'null') {
+    value = 'null';
+  } else if (type === 'undefined') {
+    value = 'undefined';
+  } else if (value === Empty) {
+    type = 'empty';
+    value = 'Empty';
+  } else {
+    value = JSON.stringify(value);
+  }
+
+  return (
+    <div class="line">
+      {(key || !key && key === 0) && <><span class={`key ${dim ? 'key-dim' : ''}`.trim()}>{key}</span><span class="sep">:</span></>}
+      <span class={`value value-${type}`}>
+        {value}
+      </span>
+    </div>
+  )
+}
+
+const Node = ({ node, key, ...props }) => {
+  const type = getType(node);
+
+  if (type === 'array' || type === 'object') {
+    let children = [];
+    if (type === 'array' && node.length > 0 && Object.entries(node).length === 0) {
+      children = Array.from({ length: node.length }, (_, key) => <Node node={Empty} key={key} />);
+    } else {
+      children = Object.entries(node).map(([key, value]) => <Node node={value} key={key} />);
+    }
+    return (
+      <Details node={node} key={key} children={children} />
+    );
+  } else if (type === 'function') {
+    return (
+      <Details node={node} key={key} children={
+      <>
+        <KeyValue key="name" value={node.name} dim={true} />
+        <KeyValue key="__proto__" value="Function" dim={true} />
+      </>
+      }/>
+    );
+  } else if (type === 'promise') {
+    return (
+      <Details node={node} key={key} children={
+        <KeyValue key="__proto__" value="Promise" dim={true} />
+      } />
+    );
+  }
+
+  return <KeyValue key={key} value={node} />;
+}
+---
+
+<div class="debug">
+  <header>
```

Comment:
Shouldn't there only be one `<header>` and `<main>`  per page? Maybe it could be a `<section>`? I'm not 100% sure, but `<main>` feels wrong here to me. This looks amazing otherwise!!!

Reply:
Good point! I'm fine just using `div`s here.

Reply:
How about 
```html
<article class="debug">
  <header> ...
  ```
I think `<article>` or `<section>` defining this as a self-contained bit makes sense, and in that context, `<header>` makes sense.